core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/window/ranking/DenseRankFunction.java

```diff
@@ -30,7 +30,7 @@ public DenseRankFunction() {
   }
 
   @Override
-  protected int rank(CumulativeWindowFrame frame) {
+  protected int rank(CurrentRowWindowFrame frame) {
```

Comment:
Could we use PeerRowsWindowFrame in here?

Reply:
I just dived into the code and it looks doable. Conceptually, `RANK` and `DENSE_RANK` share same meaning of "current row" and thus can use peer window frame. But I think we'd better use this minimal window frame for now to avoid potential data preload. We can revisit and refactor this later if necessary. Thanks!